packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -15,6 +15,7 @@ use crate::{
 };
 
 /// A Postgres-backed Datastore
+#[derive(Clone)]
```

Comment:
Do we have a way to get around this? `Clone` typically is a deep copy and having this on a `Database` feels really wrong.

Reply:
We could add the datastore as an `Arc` instead and get around this, thoughts on just doing that? We don't need to do any locking as the connections are pooled and polled for

Reply:
I wonder if we want to enforce inner mutability for `Datastore` (making all functions `&self`). On the `Allocator` trait we chose that route and it worked pretty well. We could keep it as it is currently I think. However, I see issues as soon as we have a temporary `DataStore`. Copying it would imply copying all data.